classes/migration/upgrade/v3_4_0/I7191_SubmissionProgressType.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+/**
+ * @file classes/migration/upgrade/v3_4_0/I7191_SubmissionProgressType.inc.php
+ *
+ * Copyright (c) 2014-2021 Simon Fraser University
+ * Copyright (c) 2000-2021 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class I7191_SubmissionProgressType
+ * @brief Change the submission_progress setting from an int to a string to match
+ *   the new step ids
+ */
+
+namespace PKP\migration\upgrade\v3_4_0;
+
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Schema;
+
+class I7191_SubmissionProgressType extends \PKP\migration\Migration
+{
+    public function up(): void
+    {
+        Schema::table('submissions', function (Blueprint $table) {
+            $table->text('submission_progress')->default('files')->change();
```

Comment:
I was getting an exception when running the updates here that suggested that the column type hadn't changed:
```php
ERROR: Upgrade failed: DB: SQLSTATE[22007]: Invalid datetime format: 1292 Truncated incorrect DOUBLE value: '' (SQL: update `submissions` set `submission_progress` = files where `submission_progress` = 1)
```
Oddly, checking in the MySQL client, I got a double type as expected, so there's some kind of "sure-it's-changed-but-not-quite-yet" situation going on.

I wasn't able to track down the cause, but I was able to work around it by re-using the new column creation you've done below on the `down()` function:

```php
diff --git a/classes/migration/upgrade/v3_4_0/I7191_SubmissionProgressType.php b/classes/migration/upgrade/v3_4_0/I7191_SubmissionProgressType.php
index c37a6906d5..6da3d869fc 100644
--- a/classes/migration/upgrade/v3_4_0/I7191_SubmissionProgressType.php
+++ b/classes/migration/upgrade/v3_4_0/I7191_SubmissionProgressType.php
@@ -23,16 +23,21 @@ class I7191_SubmissionProgressType extends \PKP\migration\Migration
     public function up(): void
     {
         Schema::table('submissions', function (Blueprint $table) {
-            $table->text('submission_progress')->default('files')->change();
+            $table->text('submission_progress_temp')->default('files');
         });
 
         foreach ($this->getStepMap() as $oldValue => $newValue) {
             DB::table('submissions')
                 ->where('submission_progress', $oldValue)
                 ->update([
-                    'submission_progress' => $newValue,
+                    'submission_progress_temp' => $newValue,
                 ]);
         }
+
+        Schema::table('submissions', function (Blueprint $table) {
+            $table->dropColumn('submission_progress');
+            $table->renameColumn('submission_progress_temp', 'submission_progress');
+        });
     }
 
     public function down(): void
```

Since this is what you're using in `down()` anyway, I'd suggest we just apply this change and move on.

Reply:
This didn't 100% solve the problem as `doctrine/dbal` was renaming the column using e.g...
```sql
ALTER TABLE submissions CHANGE submission_progress_temp submission_progress TEXT NOT NULL
```
...which has the unfortunate side-effect of dropping e.g. the `DEFAULT 'files'` clause, causing errors on new article creation after an upgrade. Luckily this appears to have been fixed in a newer release of `doctrine/dbal`, and updating it solves the `DEFAULT` problem -- but not the need for the patch above.

So I recommend committing the patch above, plus also [updating `doctrine/dbal`](https://github.com/NateWr/pkp-lib/pull/11/files).

Reply:
Thanks!